Error while installing RustPython on CentOS 7

```
cargo --version
cargo 1.54.0 (5ae8d74b3 2021-06-22)

rustc --version
rustc 1.54.0 (a178d0322 2021-07-26)
```

```
cargo install --git https://RustPython/RustPython


  Updating git repository `https://github.com/RustPython/RustPython`
error: failed to parse manifest at `/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/wasm/lib/Cargo.toml`

Caused by:
  feature `edition2021` is required

  this Cargo does not support nightly features, but if you
  switch to nightly channel you can add
  `cargo-features = ["edition2021"]` to enable this feature
```

UPDATE:

New Error after updating rustc
```
rustup update

rustc --version
rustc 1.57.0 (f1edd0429 2021-11-29)

cargo --version
cargo 1.57.0 (b2e52d7ca 2021-10-21)
```

```
cargo install --git https://RustPython/RustPython


  Updating git repository `https://github.com/RustPython/RustPython`
  Installing rustpython v0.1.2 (https://github.com/RustPython/RustPython#db3b3127)
    Updating crates.io index
  Downloaded tinyvec v1.5.1
  Downloaded serde v1.0.131
  Downloaded itertools v0.10.3
  Downloaded half v1.8.2
  Downloaded ppv-lite86 v0.2.15
  Downloaded env_logger v0.9.0
  Downloaded clap v2.34.0
  Downloaded pkg-config v0.3.24
  Downloaded phf v0.10.1
  Downloaded serde_json v1.0.73
  Downloaded itoa v1.0.1
  Downloaded result-like v0.4.2
  Downloaded syn v1.0.82
  Downloaded ryu v1.0.9
  Downloaded result-like-derive v0.4.2
  Downloaded serde_derive v1.0.131
  Downloaded radium v0.7.0
  Downloaded proc-macro2 v1.0.33
  Downloaded paste v1.0.6
  Downloaded phf_generator v0.10.0
  Downloaded num-bigint v0.4.3
  Downloaded cc v1.0.72
  Downloaded syn-ext v0.3.1
  Downloaded lz4_flex v0.9.2
  Downloaded rustyline v9.1.1
  Downloaded phf_macros v0.10.0
  Downloaded phf_shared v0.10.0
  Downloaded textwrap v0.14.2
  Downloaded libc v0.2.112
  Downloaded hexf-parse v0.2.1
  Downloaded fd-lock v3.0.2
  Downloaded memoffset v0.6.5
  Downloaded crc32fast v1.3.0
  Downloaded optional v0.5.0
  Downloaded 34 crates (2.1 MB) in 1.81s
   Compiling autocfg v1.0.1
   Compiling unicode-xid v0.2.2
   Compiling proc-macro2 v1.0.33
   Compiling syn v1.0.82
   Compiling libc v0.2.112
   Compiling cfg-if v1.0.0
   Compiling memchr v2.4.1
   Compiling serde v1.0.131
   Compiling serde_derive v1.0.131
   Compiling version_check v0.9.3
   Compiling parking_lot_core v0.8.5
   Compiling siphasher v0.3.7
   Compiling regex-syntax v0.6.25
   Compiling lazy_static v1.4.0
   Compiling radium v0.7.0
   Compiling scopeguard v1.1.0
   Compiling either v1.6.1
   Compiling unic-common v0.9.0
   Compiling ppv-lite86 v0.2.15
   Compiling unic-char-range v0.9.0
   Compiling log v0.4.14
   Compiling hashbrown v0.11.2
   Compiling static_assertions v1.1.0
   Compiling crunchy v0.2.2
   Compiling smallvec v1.7.0
   Compiling tiny-keccak v2.0.2
   Compiling fixedbitset v0.2.0
   Compiling new_debug_unreachable v1.0.4
   Compiling precomputed-hash v0.1.1
   Compiling bit-vec v0.6.3
   Compiling proc-macro-hack v0.5.19
   Compiling typenum v1.14.0
   Compiling pico-args v0.4.2
   Compiling diff v0.1.12
   Compiling bitflags v1.3.2
   Compiling matches v0.1.9
   Compiling once_cell v1.8.0
   Compiling hexf-parse v0.2.1
   Compiling volatile v0.3.0
   Compiling ascii v1.0.0
   Compiling semver v1.0.4
   Compiling cc v1.0.72
   Compiling regex-automata v0.1.10
   Compiling pkg-config v0.3.24
   Compiling opaque-debug v0.3.0
   Compiling block-padding v0.2.1
   Compiling tinyvec_macros v0.1.0
   Compiling serde_json v1.0.73
   Compiling unicode-width v0.1.9
   Compiling unicode_names2 v0.4.0
   Compiling crc32fast v1.3.0
   Compiling ryu v1.0.9
   Compiling unicode-segmentation v1.8.0
   Compiling itoa v1.0.1
   Compiling lalrpop-util v0.19.6
   Compiling endian-type v0.1.2
   Compiling adler v1.0.2
   Compiling crossbeam-utils v0.8.5
   Compiling rustpython-pylib v0.1.0 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/vm/pylib-crate)
   Compiling textwrap v0.14.2
   Compiling cpufeatures v0.2.1
   Compiling maplit v1.0.2
   Compiling subtle v2.4.1
   Compiling utf8parse v0.2.0
   Compiling paste v1.0.6
   Compiling optional v0.5.0
   Compiling hex v0.4.3
   Compiling strum v0.21.0
   Compiling rustpython-stdlib v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/stdlib)
   Compiling half v1.8.2
   Compiling timsort v0.1.2
   Compiling exitcode v1.1.2
   Compiling adler32 v1.2.0
   Compiling keccak v0.1.0
   Compiling unicode-casing v0.1.0
   Compiling xml-rs v0.8.4
   Compiling strsim v0.8.0
   Compiling termcolor v1.1.2
   Compiling base64 v0.13.0
   Compiling ansi_term v0.12.1
   Compiling vec_map v0.8.2
   Compiling puruspe v0.1.5
   Compiling instant v0.1.12
   Compiling lexical-util v0.8.1
   Compiling twox-hash v1.6.1
   Compiling lock_api v0.4.5
   Compiling unic-ucd-version v0.9.0
   Compiling unic-char-property v0.9.0
   Compiling itertools v0.10.3
   Compiling phf_shared v0.8.0
   Compiling phf_shared v0.10.0
   Compiling nibble_vec v0.1.0
   Compiling bit-set v0.5.2
   Compiling thread_local v1.1.3
   Compiling num-traits v0.2.14
   Compiling num-bigint v0.4.3
   Compiling indexmap v1.7.0
   Compiling num-integer v0.1.44
   Compiling memoffset v0.6.5
   Compiling miniz_oxide v0.4.4
   Compiling num-rational v0.4.0
   Compiling ahash v0.7.6
   Compiling generic-array v0.14.4
   Compiling tinyvec v1.5.1
   Compiling textwrap v0.11.0
   Compiling heck v0.3.3
   Compiling unic-ucd-hangul v0.9.0
   Compiling unic-ucd-category v0.9.0
   Compiling unic-ucd-ident v0.9.0
   Compiling unic-emoji-char v0.9.0
   Compiling unic-ucd-bidi v0.9.0
   Compiling unic-ucd-age v0.9.0
   Compiling lz4_flex v0.9.2
   Compiling radix_trie v0.2.1
   Compiling unic-ucd-normal v0.9.0
   Compiling ena v0.14.0
   Compiling libz-sys v1.1.3
   Compiling aho-corasick v0.7.18
   Compiling bstr v0.2.17
   Compiling lexical-parse-integer v0.8.0
   Compiling csv-core v0.1.10
   Compiling quote v1.0.10
   Compiling rustc_version v0.4.0
   Compiling unicode-normalization v0.1.19
   Compiling lexical-parse-float v0.8.2
   Compiling getrandom v0.2.3
   Compiling dirs-sys-next v0.1.2
   Compiling atty v0.2.14
   Compiling rand_core v0.6.3
   Compiling dirs-next v2.0.0
   Compiling petgraph v0.5.1
   Compiling rustpython-vm v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/vm)
   Compiling parking_lot v0.11.2
   Compiling term v0.7.0
   Compiling regex v1.5.4
   Compiling rand_chacha v0.3.1
   Compiling unic-normal v0.9.0
   Compiling string_cache v0.8.2
   Compiling nix v0.23.0
   Compiling time v0.1.43
   Compiling fd-lock v3.0.2
   Compiling which v4.2.2
   Compiling num_cpus v1.13.0
   Compiling uname v0.1.1
   Compiling socket2 v0.4.2
   Compiling termios v0.3.3
   Compiling gethostname v0.2.1
   Compiling digest v0.9.0
   Compiling block-buffer v0.9.0
   Compiling crypto-mac v0.8.0
   Compiling ascii-canvas v3.0.0
   Compiling rand v0.8.4
   Compiling clap v2.34.0
   Compiling env_logger v0.9.0
   Compiling flate2 v1.0.22
   Compiling blake2 v0.9.2
   Compiling sha3 v0.9.1
   Compiling sha2 v0.9.8
   Compiling sha-1 v0.9.8
   Compiling md-5 v0.9.1
   Compiling chrono v0.4.19
   Compiling dns-lookup v1.0.8
   Compiling mt19937 v2.0.1
   Compiling Inflector v0.11.4
   Compiling phf_generator v0.10.0
   Compiling lalrpop v0.19.6
   Compiling caseless v0.2.1
   Compiling rustyline v9.1.1
   Compiling syn-ext v0.3.1
   Compiling pmutil v0.5.3
   Compiling thiserror-impl v1.0.30
   Compiling phf_macros v0.10.0
   Compiling derivative v2.2.0
   Compiling result-like-derive v0.4.2
   Compiling is-macro v0.1.9
   Compiling strum_macros v0.21.1
   Compiling phf v0.10.1
   Compiling thiserror v1.0.30
   Compiling result-like v0.4.2
   Compiling rustpython-parser v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/parser)
   Compiling num-complex v0.4.0
   Compiling bincode v1.3.3
   Compiling toml v0.5.8
   Compiling proc-macro-crate v1.1.0
   Compiling rustpython-common v0.0.0 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/common)
   Compiling rustpython-bytecode v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/bytecode)
   Compiling num_enum_derive v0.5.4
   Compiling rustpython-ast v0.1.0 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/ast)
   Compiling num_enum v0.5.4
   Compiling sre-engine v0.1.2
   Compiling rustpython-compiler-core v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/compiler)
   Compiling rustpython-compiler v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/compiler/porcelain)
   Compiling rustpython-derive v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312/derive)
   Compiling rustpython v0.1.2 (/home/hamza2/.cargo/git/checkouts/rustpython-f8ef4d934ac33cd8/db3b312)
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.0.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.1.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.10.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.11.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.12.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.13.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.14.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.15.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.2.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.3.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.4.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.5.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.6.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.7.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.8.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.rustpython.239d8340-cgu.9.rcgu.o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49.1iyuc9q5ulswclvk.rcgu.o" "-Wl,--as-needed" "-L" "/tmp/cargo-installGhYVs7/release/deps" "-L" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/cargo-installGhYVs7/release/deps/librustpython-1dfcb7dcc3fde621.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_stdlib-ce7191cc72a7383b.rlib" "/tmp/cargo-installGhYVs7/release/deps/libdns_lookup-07721dd1b522b43b.rlib" "/tmp/cargo-installGhYVs7/release/deps/libgethostname-f8974313e114a8c9.rlib" "/tmp/cargo-installGhYVs7/release/deps/libmt19937-6ae2491181ae2208.rlib" "/tmp/cargo-installGhYVs7/release/deps/libpuruspe-889c1b598fa86d22.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcsv_core-2ab8ce2b0d924db8.rlib" "/tmp/cargo-installGhYVs7/release/deps/libbase64-9689bc046dcfffbd.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtermios-8ddf04ce21addf7a.rlib" "/tmp/cargo-installGhYVs7/release/deps/libflate2-bb683de52e8aff31.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblibz_sys-3beba3d8667574e5.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcrc32fast-3510d4280a84e716.rlib" "/tmp/cargo-installGhYVs7/release/deps/libadler32-d853300f1018cd22.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_age-d99439a6a8d5eabb.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_normal-d4edaa088c036d79.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_normal-bd81f5b524d1b01a.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_hangul-8cc88f7da7fb50f9.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsocket2-7a1c845444d6d173.rlib" "/tmp/cargo-installGhYVs7/release/deps/libxml-996e9bad523e34e2.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsha3-690fdf9d10e0c3d0.rlib" "/tmp/cargo-installGhYVs7/release/deps/libkeccak-5b046aed3a1e9aa2.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsha2-c64b9ff0685d383c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsha1-111b8bb1696ee4f9.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcpufeatures-3c65ce97caa7efc6.rlib" "/tmp/cargo-installGhYVs7/release/deps/libmd5-097c84d7e533ebf8.rlib" "/tmp/cargo-installGhYVs7/release/deps/libblock_buffer-67a0432477463236.rlib" "/tmp/cargo-installGhYVs7/release/deps/libblock_padding-acd205a7620de1c4.rlib" "/tmp/cargo-installGhYVs7/release/deps/libblake2-8a5fbc2ef6719f2a.rlib" "/tmp/cargo-installGhYVs7/release/deps/libopaque_debug-5532a40e8e391b1d.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcrypto_mac-811933217d2a661e.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsubtle-57e99c68af6138ed.rlib" "/tmp/cargo-installGhYVs7/release/deps/libdigest-fb49b90626b7f715.rlib" "/tmp/cargo-installGhYVs7/release/deps/libgeneric_array-b1761c6abc3f6214.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtypenum-8f37907d38043bf5.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_vm-0a3b932f8030b015.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_pylib-7855d1cd085a55b3.rlib" "/tmp/cargo-installGhYVs7/release/deps/libwhich-e98eb708b667aabf.rlib" "/tmp/cargo-installGhYVs7/release/deps/libstrum-639e86b555cd4274.rlib" "/tmp/cargo-installGhYVs7/release/deps/libuname-e488d161153ad6b1.rlib" "/tmp/cargo-installGhYVs7/release/deps/libexitcode-2fbc5b0a65ec74b7.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_cpus-9808194b5f7a4133.rlib" "/tmp/cargo-installGhYVs7/release/deps/libhex-968c02a3cc0ae5d2.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcaseless-6f53acb81c342085.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunicode_normalization-b439cc1a975af915.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtinyvec-575ab6343979d7d0.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtinyvec_macros-133174a81e9385d3.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtimsort-87cc8b94cc27dc90.rlib" "/tmp/cargo-installGhYVs7/release/deps/libthread_local-023a7999a3ca0e50.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsre_engine-0b7d823a47e2c213.rlib" "/tmp/cargo-installGhYVs7/release/deps/libhalf-0f2ff24b33eca874.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_enum-d0775e568998eef6.rlib" "/tmp/cargo-installGhYVs7/release/deps/libresult_like-9e5fbc42c7035e0d.rlib" "/tmp/cargo-installGhYVs7/release/deps/libchrono-134a4c3687f790db.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtime-d3fa51d9cfd7da72.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_compiler-ac9d07e453b2893c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libthiserror-bac45699dc7b6d70.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_parser-97c17029caca0f58.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunicode_names2-9d31d99f6417cb5c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libphf-c61704ff4302acc3.rlib" "/tmp/cargo-installGhYVs7/release/deps/libphf_shared-0dbed7180c8934a5.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_emoji_char-571d6ab86cf351e6.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblalrpop_util-eb028c013ce738ea.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_compiler_core-d86495b7327ee974.rlib" "/tmp/cargo-installGhYVs7/release/deps/libahash-7d6355b9ea1aa2b7.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_bytecode-e81f49762ec0c901.rlib" "/tmp/cargo-installGhYVs7/release/deps/libbincode-dee6a7762d42e316.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblz4_flex-76024c3a44d498fb.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtwox_hash-5227abfb356588f8.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_ast-a3436a3c0216ebeb.rlib" "/tmp/cargo-installGhYVs7/release/deps/liboptional-13b40a01d1e68991.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustyline-51e3929f41555170.rlib" "/tmp/cargo-installGhYVs7/release/deps/libutf8parse-45099463e2f9faab.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnix-c4c5f13d9a18dc40.rlib" "/tmp/cargo-installGhYVs7/release/deps/libmemoffset-14f343850be02c38.rlib" "/tmp/cargo-installGhYVs7/release/deps/libfd_lock-fdcdfa4f2ffc011d.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunicode_segmentation-e000ecd89a23d67c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libdirs_next-43fc267404459163.rlib" "/tmp/cargo-installGhYVs7/release/deps/libdirs_sys_next-3c75ff3b2c3eac85.rlib" "/tmp/cargo-installGhYVs7/release/deps/libradix_trie-35cbd015634d2ae2.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnibble_vec-027305d325cc0703.rlib" "/tmp/cargo-installGhYVs7/release/deps/libendian_type-10db0d0fa5216ec3.rlib" "/tmp/cargo-installGhYVs7/release/deps/libindexmap-41e5f75c0a8c3e53.rlib" "/tmp/cargo-installGhYVs7/release/deps/libhashbrown-e61bbb41678c0d80.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunicode_casing-380a77f580b3b63c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_ident-1be09ed58be9dfb1.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_bidi-a77e04b9050bd6ee.rlib" "/tmp/cargo-installGhYVs7/release/deps/libitertools-7593e6d4f3b8d64f.rlib" "/tmp/cargo-installGhYVs7/release/deps/libeither-ee05c1c2c41537bc.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_rational-e67f7da051f121e2.rlib" "/tmp/cargo-installGhYVs7/release/deps/librustpython_common-22cc1edba2a2aaa0.rlib" "/tmp/cargo-installGhYVs7/release/deps/librand-84fe2f9b03375b5d.rlib" "/tmp/cargo-installGhYVs7/release/deps/librand_chacha-841fcd24f9bd1612.rlib" "/tmp/cargo-installGhYVs7/release/deps/libppv_lite86-b4dbb70e1e0a5c67.rlib" "/tmp/cargo-installGhYVs7/release/deps/librand_core-c3466ea4d9c564c7.rlib" "/tmp/cargo-installGhYVs7/release/deps/libgetrandom-913e00cf33a9f22c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libhexf_parse-9866003a0e3b6eb3.rlib" "/tmp/cargo-installGhYVs7/release/deps/libparking_lot-0be25a7eec312e09.rlib" "/tmp/cargo-installGhYVs7/release/deps/libparking_lot_core-1b41ca8006c4648b.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsmallvec-dcaec7630626cd4e.rlib" "/tmp/cargo-installGhYVs7/release/deps/libinstant-a4cfb1b0b416202a.rlib" "/tmp/cargo-installGhYVs7/release/deps/libonce_cell-9574b8414d913f8c.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblock_api-966d5e1d2a90b4f7.rlib" "/tmp/cargo-installGhYVs7/release/deps/libscopeguard-c857144e7f0e1819.rlib" "/tmp/cargo-installGhYVs7/release/deps/libsiphasher-c1f19f7360f63a65.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblexical_parse_float-8ae93b9279df148d.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblexical_parse_integer-e26f58ad2ed8cd00.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblexical_util-b63b7743c6197f9c.rlib" "/tmp/cargo-installGhYVs7/release/deps/libstatic_assertions-af68a8dafc5b3a27.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_bigint-4ca9ce8df70ec62f.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_integer-38dcfbcaa0f2a117.rlib" "/tmp/cargo-installGhYVs7/release/deps/libascii-68e61fa58fa4d7b8.rlib" "/tmp/cargo-installGhYVs7/release/deps/libvolatile-36e7a36d56d34181.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_category-504530a6d4e4b222.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_ucd_version-3c14c854939fcf50.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_common-ddb8595c9bdd6666.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_char_property-623ee952c6cc2ea3.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunic_char_range-e56c67a015d66735.rlib" "/tmp/cargo-installGhYVs7/release/deps/libmatches-44f7ddad63674e12.rlib" "/tmp/cargo-installGhYVs7/release/deps/libradium-7ffe01bff7cc4096.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_complex-db25deaf0e8dde40.rlib" "/tmp/cargo-installGhYVs7/release/deps/libserde-cf596fa4a7d4c847.rlib" "/tmp/cargo-installGhYVs7/release/deps/libbstr-420a122de7d84c6b.rlib" "/tmp/cargo-installGhYVs7/release/deps/libregex_automata-56a89e95823b9b2b.rlib" "/tmp/cargo-installGhYVs7/release/deps/libmemchr-06d76bcb96a352b9.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcrossbeam_utils-eadf027b921bbccf.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblazy_static-0df5d4f37a836555.rlib" "/tmp/cargo-installGhYVs7/release/deps/libnum_traits-b21d207a8542564a.rlib" "/tmp/cargo-installGhYVs7/release/deps/libenv_logger-6a4be823ef736ec9.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtermcolor-c1e591ee2a17ad46.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblog-f8752b7be0ee0450.rlib" "/tmp/cargo-installGhYVs7/release/deps/libcfg_if-a04351a1e37ac403.rlib" "/tmp/cargo-installGhYVs7/release/deps/libclap-6a30770dd8252282.rlib" "/tmp/cargo-installGhYVs7/release/deps/libvec_map-5d3b872e0a1408b5.rlib" "/tmp/cargo-installGhYVs7/release/deps/libtextwrap-c273469f9debde5a.rlib" "/tmp/cargo-installGhYVs7/release/deps/libunicode_width-b88877dbb8531487.rlib" "/tmp/cargo-installGhYVs7/release/deps/libstrsim-65d231e2011670e5.rlib" "/tmp/cargo-installGhYVs7/release/deps/libbitflags-045e5e9961c5ccd8.rlib" "/tmp/cargo-installGhYVs7/release/deps/libatty-3779a6842c7e382f.rlib" "/tmp/cargo-installGhYVs7/release/deps/liblibc-0540ccfb702360f2.rlib" "/tmp/cargo-installGhYVs7/release/deps/libansi_term-594c980194a3606c.rlib" "-Wl,--start-group" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-7c582493123fc1dd.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-1392776590706175.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-2eb6edf4d031cd1e.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-33a7ad3b5f7fedf6.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-7d32adce541987d9.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-5cb369120f224726.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-6ab2efd1d2f431a9.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-567f611439253c7e.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-ebbc63efd6d2efc5.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-862830f0d224a2e1.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-cad0401ae7a80e32.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-fd54290077194763.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-4fc3313c8ccb1ec0.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-19bf8dffe82b09d4.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-afc95b1640c4beca.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-aff6658baa87e3d1.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-8be8a1689a4f7b48.rlib" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-5284934f66073844.rlib" "-Wl,--end-group" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-2a0b2a4f96acb821.rlib" "-Wl,-Bdynamic" "-lc" "-lz" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/home/hamza2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/tmp/cargo-installGhYVs7/release/deps/rustpython-385d88b3a8d18b49" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs"
  = note: /tmp/cargo-installGhYVs7/release/deps/librustpython_vm-0a3b932f8030b015.rlib(rustpython_vm-0a3b932f8030b015.rustpython_vm.2328e824-cgu.15.rcgu.o): In function `rustpython_vm::function::IntoPyNativeFunc::into_func::_$u7b$$u7b$closure$u7d$$u7d$::h1eebc6a6b772d7ea':
          rustpython_vm.2328e824-cgu.15:(.text._ZN13rustpython_vm8function16IntoPyNativeFunc9into_func28_$u7b$$u7b$closure$u7d$$u7d$17h1eebc6a6b772d7eaE+0x17f): undefined reference to `getrandom'
          collect2: error: ld returned 1 exit status
          
  = help: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: failed to compile `rustpython v0.1.2 (https://github.com/RustPython/RustPython#db3b3127)`, intermediate artifacts can be found at `/tmp/cargo-installGhYVs7`

Caused by:
  could not compile `rustpython` due to previous error
```

--- a/vm/src/stdlib/posix.rs
+++ b/vm/src/stdlib/posix.rs
@@ -1981,13 +1981,18 @@ pub mod module {
     }
 
     #[cfg(target_os = "linux")]
+    unsafe fn sys_getrandom(buf: *mut libc::c_void, buflen: usize, flags: u32) -> isize {
+        libc::syscall(libc::SYS_getrandom, buf, buflen, flags as usize) as _
+    }
+
+    #[cfg(target_os = "linux")]
     #[pyfunction]
     fn getrandom(size: isize, flags: OptionalArg<u32>, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
         let size = usize::try_from(size)
             .map_err(|_| vm.new_os_error(format!("Invalid argument for size: {}", size)))?;
         let mut buf = Vec::with_capacity(size);
         unsafe {
-            let len = libc::getrandom(
+            let len = sys_getrandom(
                 buf.as_mut_ptr() as *mut libc::c_void,
                 size,
                 flags.unwrap_or(0),

